### PR TITLE
Update the user guide to correct typos and include validation procedures used in the system.

### LINKED ISSUE #337: [PE-D][Tester D] Restrictions on name format sometines hinder usage
# Replication
- execute `add n/Ramesh Kumar s/o Prabhakar Reddy Iyer p/99999999 e/user@example.com a/ABC town`

# Actual hbehavior
An error regarding name format is shown.

# Expected
No errors should be shown since `s/o` is a valid name in some countries.

# Suggestion

I suggest we should loosen the restriction on the name as well. It might be a good idea to just allow any arbitrary character sequences since a name can contain special characters.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: lithops-zty/ped#6

### PR DESCRIPTION
**Pull request by sheenkerr**
Closes #337, #335, #327 & #352

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -85,10 +85,16 @@ Adds a person to the address book.
...
+<div markdown="span" class="alert alert-warning">
+**Names:**
 Names must be alphanumeric and all contacts must have unique names.
+Names with symbols like "s/o" and "d/o" will not be allowed. Please write names without these symbols.
+Names with additional white spaces in between words will be taken as it is.
 
 The application will throw an error if you attempt to create two users with the same names
+
+**Phone Numbers:**
+Only Singapore phone numbers are allowed by the application. Meaning, that only phone numbers starting with the number 6, 8 or 9 and is exactly 8 digits in total is allowed.
```
Comment (ITLimJiaWei) [inline docs/UserGuide.md:97]: Small grammatical mistake "with the number 6, 8 or 9 and **are** exactly 8 digits in total **are** allowed". Though we may drop this feature so TBD

**Review by ITLimJiaWei — CHANGES_REQUESTED** [review]
As commented

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -648,6 +648,32 @@ _Similar to UC10 except without extension 2b._
...
+
+    - 2a1. System shows an error message.
+
+      Use case ends.
+
+- 2b. The details entered are invalid.
+
+    - 2b1. System shows an error message.
+
+      Use case ends.
```
Comment (Nicholas-Cheng-De-Fei) [inline docs/DeveloperGuide.md:674]: I believe you are missing extension when use inputs a empty field.

Example `edit 1 p/  a/`

File: docs/UserGuide.md [hunk 3]
```diff
@@ -359,6 +366,15 @@ Examples:
 
 - `toggle` changes the application theme.
 
+### Navigating Command History : 
+
+Easily cycle through your previous **successfully executed** commands by using the Up and Down arrow keys on your keyboard.
+
+Format: `Up arrow`
```
Comment (Nicholas-Cheng-De-Fei) [inline docs/UserGuide.md:373]: Maybe specify on the keyboard and not in the command box later some people just type `up arrow`, then they see invalid command.

**Review by Nicholas-Cheng-De-Fei — CHANGES_REQUESTED** [review]

File: docs/UserGuide.md (hunk 1 above)
Comment (ITLimJiaWei) [reply docs/UserGuide.md:97]: Update: As suggested by Kenneth, can leave this in pending the grammatical mistakes being fixed and specify that international number support is coming in a future feature. "While we are aware of the possibility of international numbers, for this iteration, we decided to focus on the local population"

File: docs/UserGuide.md [hunk 4]
```diff
@@ -384,6 +400,13 @@ If your changes to the data file makes its format invalid, AddressBook will disc
 Furthermore, certain edits can cause the AddressBook to behave in unexpected ways (e.g., if a value entered is outside the acceptable range). Therefore, edit the data file only if you are confident that you can update it correctly.
 </div>
 
+---
+
+## Future Features
+
+1) While we are aware of the possibility of international numbers, for this iteration, we decided to focus on the local population.
```
Comment (ITLimJiaWei) [inline docs/UserGuide.md:407]: Could you explicitly state support for international numbers is planned? E.g. "Support for these numbers is planned in a future release." I know the current statement implies it but I think better to state it clearly.

**Review by ITLimJiaWei — CHANGES_REQUESTED** [review]

**Review by ITLimJiaWei — APPROVED** [review]
LGTM

**Review by Nicholas-Cheng-De-Fei — APPROVED** [review]
LGTM

**Pull request merged**